src/QtComponents/QtMgx3DMainWindow.cpp

```diff
@@ -5255,8 +5254,49 @@ log (t4);
+	const TkUtil::Color	currentTopoColorWith0DProj (Internal::InternalPreferences::instance ( )._topoColorWith0DProj.getRed ( ), Internal::InternalPreferences::instance ( )._topoColorWith0DProj.getGreen ( ), Internal::InternalPreferences::instance ( )._topoColorWith0DProj.getBlue ( ));;
+	const TkUtil::Color	currentTopoColorWith1DProj (Internal::InternalPreferences::instance ( )._topoColorWith1DProj.getRed ( ), Internal::InternalPreferences::instance ( )._topoColorWith1DProj.getGreen ( ), Internal::InternalPreferences::instance ( )._topoColorWith1DProj.getBlue ( ));;
+	const TkUtil::Color	currentTopoColorWith2DProj (Internal::InternalPreferences::instance ( )._topoColorWith2DProj.getRed ( ), Internal::InternalPreferences::instance ( )._topoColorWith1DProj.getGreen ( ), Internal::InternalPreferences::instance ( )._topoColorWith2DProj.getBlue ( ));;
+	const TkUtil::Color	currentTopoColorWith3DProj (Internal::InternalPreferences::instance ( )._topoColorWith3DProj.getRed ( ), Internal::InternalPreferences::instance ( )._topoColorWith3DProj.getGreen ( ), Internal::InternalPreferences::instance ( )._topoColorWith3DProj.getBlue ( ));;
+	const bool	updateAll	= currentTopoColorWithoutProj != oldTopoColorWithoutProj ? true : false;
+	const bool	update0D	= (true == updateAll) || (currentTopoColorWith0DProj != oldTopoColorWith0DProj) ? true : false;
+	const bool	update1D	= (true == updateAll) || (currentTopoColorWith1DProj != oldTopoColorWith1DProj) ? true : false;
+	const bool	update2D	= (true == updateAll) || (currentTopoColorWith2DProj != oldTopoColorWith2DProj) ? true : false;
+	const bool	update3D	= (true == updateAll) || (currentTopoColorWith3DProj != oldTopoColorWith3DProj) ? true : false;
+	if (true == update0D)
+	{
+		std::vector<Topo::Vertex* >	vertices;
+		getContext ( ).getTopoManager ( ).getVertices (vertices);
+		for (std::vector<Topo::Vertex*>::iterator itv = vertices.begin ( ); vertices.end ( ) != itv; itv++)
+			(*itv)->updateDisplayPropertiesColor ( );
+	}	// if (true == update0D)
+	if (true == update1D)
+	{
+		std::vector<Topo::CoEdge*>	edges;
+		getContext ( ).getTopoManager ( ).getCoEdges (edges);
+		for (std::vector<Topo::CoEdge*>::iterator ite = edges.begin ( ); edges.end ( ) != ite; ite++)
+			(*ite)->updateDisplayPropertiesColor ( );
+	}	// if (true == update1D)
+	if (true == update2D)
+	{
+		std::vector<Topo::CoFace*>	faces;
+		getContext ( ).getTopoManager ( ).getCoFaces (faces);
+		for (std::vector<Topo::CoFace*>::iterator itf = faces.begin ( ); faces.end ( ) != itf; itf++)
+			(*itf)->updateDisplayPropertiesColor ( );
+	}	// if (true == update2D)
+	if (true == update3D)
+	{
+		std::vector<Topo::Block* >	blocks;
+		getContext ( ).getTopoManager ( ).getBlocks (blocks);
+		for (std::vector<Topo::Block*>::iterator itb = blocks.begin ( ); blocks.end ( ) != itb; itb++)
+			(*itb)->updateDisplayPropertiesColor ( );
+	}	// if (true == update3D)
```

Comment:
Entities display color might not be updated in case where higher dimension color display is changed. For example if 2D projection color is changed, the 0D and 1D entities (vertices and edges) color will not be updated